python/tank/templatekey.py

```diff
@@ -328,7 +329,22 @@ def __init__(self,
         elif self._filter_by is not None:
             # filter_by is a regex
             self._custom_regex_u = re.compile(self._filter_by, re.UNICODE)
-        
+
+        self._subset_str = subset
+        self._subset_format = subset_format
+        if self._subset_format and sys.version_info < (2, 6):
+            raise TankError("Subset formatting in template keys require python 2.6+!")
+
+        self._subset_str = subset
+
+        if subset:
+            try:
+                self._subset_regex = re.compile(subset, re.UNICODE)
+            except Exception, e:
```

Comment:
[minor] Now that we're deprecating support for Python 2.5, do we want to make a shift over to using `except Exception as e:` instead of the old-style syntax? There are issues with the old style when you get into catching multiple exception types.